spi: stm32: Set behavior 16bits tx in 8 bits mode optional

erwango opened the pull request:
commit 9986c18
Author: Erwan Gouriou <erwan.gouriou@linaro.org>
Date:   Mon Jan 6 14:26:43 2020 +0100
```
    spi: stm32: Fixes to shift frame function refactoring
    
    Refactoring of function spi_stm32_shift_m8/16 in PR #20948 which
    introduced optional behavior "Allow 16bit sends in 8bit mode if
    possible in master-only mode" has not been done with respecting
    previous function behavior.
    
    Fix newly created functions to respect flow of previous code:
    -Do not impact spi context before tx empty condition is verified
    -Update tx context and get tx_frame under spi_context_tx_on condition
    (not spi_context_tx_buf_on condition)
    -Always update rx context after data reception, not only under
    spi_context_rx_buf_on condition
    
    Last fix is known to fix regression on some cases.
    Two first changes may be conservative and revisited individually
    later on, aim is to minimize the changes in a first iteration.
    
    Fixes #21679
    
    Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
```

commit 514d1cb
Author: Erwan Gouriou <erwan.gouriou@linaro.org>
Date:   Mon Jan 6 09:44:13 2020 +0100
```
    spi: stm32: Set behavior 16bits tx in 8 bits mode optional
    
    Behavior of sending 16bits in 8bits mode when possible is not working
    with all slaves, as it appears.
    should be
    It should be optional and not be active by default.
    
    Fixes #21546
    Fixes #21679
    
    Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
```

erwango committed:
spi: stm32: Set behavior 16bits tx in 8 bits mode optional

Behavior of sending 16bits in 8bits mode when possible is not working
with all slaves, as it appears.
should be
It should be optional and not be active by default.

Fixes #21546
Fixes #21679

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@Mani-Sadhasivam, @StefJar

erwango commented:
@psidhu

StefJar commented:
I tested the it, but it does not fix my issue https://github.com/zephyrproject-rtos/zephyr/issues/21679

erwango committed:
spi: stm32: Fixes to shift frame function refactoring

Refactoring of function spi_stm32_shift_m8/16 in PR #20948 which
introduced optional behavior "Allow 16bit sends in 8bit mode if
possible in master-only mode" has not been done with respecting
previous function behavior.

Fix newly created functions to respect flow of previous code:
-Do not impact spi context before tx empty condition is verified
-Update tx context and get tx_frame under spi_context_tx_on condition
(not spi_context_tx_buf_on condition)
-Always update rx context after data reception, not only under
spi_context_rx_buf_on condition

Last fix is known to fix regression on some cases.
Two first changes may be conservative and revisited individually
later on, aim is to minimize the changes in a first iteration.

Fixes #21679

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@StefJar, I pushed an additional fix which fixes the issue on my side.
Please have a check.

erwango commented:
Closed following after https://github.com/zephyrproject-rtos/zephyr/pull/21722
